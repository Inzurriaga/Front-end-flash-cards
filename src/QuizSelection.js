import React, { Component } from 'react';
import './style/Main.scss';
import Quiz from "./quizComponent"
import Controls from "./Controls"

class StudyControls extends Component {
    constructor() {
        super();
        this.state = {
            deck: []
        }
    }

    createDeck = () => {
        let catagorys = []
        document.querySelectorAll('input[name="pick-catagory"]:checked').forEach((picked) => {
            catagorys.push(picked.value);
       });
        this.setState({ deck: catagorys })
    }
    render() {
        if(this.state.deck.length <= 1) {
        return (
            <div>
                <h2>select catagories</h2>
                <Controls create={this.createDeck}
                        selection={this.props.selection}/>
            </div>
        )}else {
            return (
            <div>
                <Quiz cards={this.props.cards}/>
            </div>
            )
        }
    }
}

export default StudyControls;