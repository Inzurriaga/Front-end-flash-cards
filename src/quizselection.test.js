import React from 'react';
import QuizSelection from './QuizSelection';
import { shallow } from 'enzyme';

describe('the quiz selection component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <QuizSelection />);
  });

  it('should display all quiz selection', () => {
    expect(wrapper).toMatchSnapshot();
  });

});