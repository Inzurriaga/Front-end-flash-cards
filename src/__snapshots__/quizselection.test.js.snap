// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the quiz selection component should display all quiz selection 1`] = `
<div>
  <h2>
    select catagories
  </h2>
  <Controls
    create={[Function]}
  />
</div>
`;
